### ISSUE TITLE
cucumber-rails: Ruby 2.7 readiness issue

### ISSUE THREAD

**Issue by olleolleolle** [cucumber/cucumber-rails#455]
## Summary

We want to be compatible with current Ruby versions. 2.7 is not yet in the matrix. It could be.

## Expected Behavior

- Have 2.7 in CI matrix
- No Ruby 2.7 warnings from the test suite
- All tests pass (or, no tests fail due to 2.7)

## Current Behavior

- Does not have Ruby 2.7 in the test matrix

## Possible Solution

- Extend the matrix.

## Context & Motivation

Rails is moving forwards with their warnings-removal work for 2.7. This is a Rails plugin. Would be nice to move in the same direction.

**Comment by luke-hill** [comment]
I would advise that this is `hard-blocked` by the issues in cucumber / cucumber-core repos.

https://github.com/cucumber/cucumber-ruby-core/issues/187
https://github.com/cucumber/cucumber/issues/843

**Comment by olleolleolle** [comment]
@luke-hill Both of these issues are closed.

**Comment by DannyBen** [comment]
Since the `cucumber-ruby` gem is in [4.0 Release Candidate state for over a year](https://github.com/cucumber/cucumber-ruby/releases), may I suggest to consider at least releasing a `cucumber-rails` RC version, with the latest and greatest `cucumber`(RC) and `cucumber-core`?

This lag is beginning to have an impact on productivity, I guess not only for me.

**Comment by luke-hill** [comment]
@DannyBen This issue is now hard blocked on #466 and I suggest once #466 is released we can probably close this as no longer relevant / fixed.

as for getting cucumber ruby 4 support in cucumber rails, This was done recently in the Open Source work. However it's anticipated some small issues might come crawling out.

**Comment by olleolleolle** [comment]
@mvz You've gotten #466 merged - and I just checked, no warnings are showing in the Job

- 5.2 https://travis-ci.org/github/cucumber/cucumber-rails/jobs/697534322
- 6.0 https://travis-ci.org/github/cucumber/cucumber-rails/jobs/697534321

Yay!
